### DISCUSSION TITLE
[Dev Tools] Using hardhat-zksync-upgradable with Solidity 0.8.18

### DISCUSSION THREAD

**Discussion by cytadela8** [zkSync-Community-Hub/zksync-developers#729] [Dev Tooling] [answered]
### Environment

Testnet

### Select the Dev Tool you are using

Hardhat Plugins

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I'm running into compilation issue with @openzeppelin/contracts-hardhat-zksync-upgradable. It seems it is incompatible with my Solidity version (0.8.18). (See screenshot 1)My questions are:

1. What is the `@openzeppelin/contracts-hardhat-zksync-upgradable` package. I struggle to find it's sources on Github or it's page on npm.
2. I think downgrading to `@openzeppelin/contracts-upgradeable` may be a solution. What version of hardhat-zksync-upgradable should I use? 1.6.0? The docs seem to have a mistake. I think it should say "For plugin version <1.7.0: Compatible with @openzeppelin/contracts-upgradeable v4." (See screenshot 2)

## Screenshot 1

![image](https://github.com/user-attachments/assets/ab7accac-64ec-4b3e-9939-0760e78e5fe8)

## Screenshot 2

![image](https://github.com/user-attachments/assets/1b432ee8-5a03-4bdf-a318-c94fbcc45c81)

**Comment by alhenaai** [comment]
### Answers to Your Questions

1. **Package Information:**
   The `@openzeppelin/contracts-hardhat-zksync-upgradable` package you mentioned seems to be a typo or confusion. The correct package for working with upgradable contracts in a Hardhat environment on ZKsync is `@matterlabs/hardhat-zksync-upgradable`. This package is designed to work with the OpenZeppelin upgrades plugins for deploying and managing upgradable smart contracts on ZKsync. You can find more details and installation instructions on the [npm page](https://www.npmjs.com/package/@matterlabs/hardhat-zksync-upgradable) and the [official documentation](https://docs.zksync.io/build/tooling/hardhat/plugins/hardhat-zksync-upgradable).

2. **Version Compatibility:**
   For using `@openzeppelin/contracts-upgradeable` with the `hardhat-zksync-upgradable` plugin:
   - If you are using a plugin version **less than 1.7.0**, you should use `@openzeppelin/contracts-upgradeable` version **4**.
   - For plugin version **1.7.0 or higher**, it is recommended to use `@openzeppelin/contracts-upgradeable` version **5**.

   It seems there was indeed a misunderstanding in the documentation you referred to. The correct compatibility information is as stated above and can be verified in the [documentation](https://docs.zksync.io/build/tooling/hardhat/plugins/hardhat-zksync-upgradable).

### Suggested Steps

- Ensure you are using the correct package name in your project.
- Adjust your plugin and OpenZeppelin contracts versions according to the compatibility guidelines provided.

If you need further assistance or clarification, feel free to ask!

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by dutterbutter** [comment, marked as answer]
Hey @cytadela8 👋 

You should make use of the `hardhat-zksync` hardhat plugin which out of the box is shipped with `hardhat-zksync-upgradable`. No need to worry about version and its compatible with 0.8.18 without issue. 

The issue you are experiencing is that the pragma in your contract is not aligned with the latest version. Please provide the contract that is 0.8.18. It should have a `^`.

**Comment by dutterbutter** [comment]
To answer your question directly the `@openzeppelin/contracts-hardhat-zksync-upgradable` is `@openzeppelin/contracts@^5.0.2` but with a separate namespace to avoid collision with the v4 version. You can see it here in the `package.json` of the zksync upgradable package: https://github.com/matter-labs/hardhat-zksync/blob/be43a8335cecbcba0d37a476ce9e3e070dfd0908/packages/hardhat-zksync-upgradable/package.json#L39C63-L39C93
